### ISSUE TITLE
Inconsistency between names in the conv 1d operations and no support for 'causal' padding

### ISSUE THREAD

**Issue by Uiuran** [tensorflow/tensorflow#29804]
only tf.keras.layers.Conv1D supports padding 'causal', and it is very important for time series research, however in

tf.python.ops.conv1d
input: (batch_size, dim1size, dim2size, ... , channels_in)
filters: (filter_size, channels_in, channels_out)

Also this function behaves as expected, setting a op node according to the math and predicting an output Tensor of predictable shape.

tf.keras.layers.Conv1D
input: (batch_size, steps, input_dim)
filter: size of 1D filter
output: (batch_size, new_steps, filters) 

These are non usual names for the mathematical definition of convolution. Is "steps" the size of the time series ? new_steps the relation of the filter size with the input_dim ? are filters the channel output size ? 

Seems incompatible with the former that follows closely the mathematical definition and also expands for multi-channel like the 2D case. Also, is desirable to use causal padding with tf.python.ops.conv1d, without having to add extraneous functions by hand to the code.

thank you, pls have your appreciation.

**Comment by ravikyram** [comment]
Can you please help us to understand  tf.python.ops.conv1d  refers to which module or function ,since the mentioned function name looks related to tf.nn.conv1d & also please provide Tensorflow version.Thanks

**Comment by Uiuran** [comment]
Exactly. However in tf.python.ops.nn_ops are where those convs are defined. It exports as nn.conv1d
It export 2 times the same thing, one with deprecation warning. 

Now i dont know where daheck* ive seem this:
 input: (batch_size, dim1size, dim2size, ... , channels_in)

But it does make sense for RMN/EEG and other biomedical datasets to do conv1d in data with more than one dimension.

Still resides the difference between the names, which is confusing.
TF version 1.13.01, conda-forge py27_0 build 

* sorry ive been sick and on fever, almost unable to code and read code.

**Comment by tensorflowbutler** [comment]
Hi There,

 This is a stale issue. As you are using an older version of tensorflow, we are checking to see if you still need help on this issue. Please test the issue with the latest TensorFlow (TF2.7 and tf-nightly). If the issue still persists with the newer versions of TF, please feel free to open it in [keras-team/keras](https://github.com/keras-team/keras/issues) repository by providing details about the issue and a standalone code to reproduce the issue. Thanks! 

 Please note that Keras development has moved to a separate Keras-team/keras repository to focus entirely on only Keras. Thanks!
